warrior/Framework/ClassUtils/gnmi_utils_class.py

```diff
@@ -0,0 +1,339 @@
+            child = data_Utils.get_object_from_datarepository(session_id)
+            child.sendline(cmd_string)
+        if script.lower() == "no":
+            u_index = child.expect(["username.*", ".*error.*", pexpect.EOF, pexpect.TIMEOUT],
+                                   timeout=5)
+            if u_index == 0:
+                child.sendline(uname)
+                status = True
+                testcase_Utils.pNote(child.match.group(0) + uname)
+                excute = True
+            else:
+                excute = False
+            p_index = child.expect(["password.*", ".*error.*", pexpect.EOF, pexpect.TIMEOUT],
+                                   timeout=5)
+            if p_index == 0:
+                testcase_Utils.pNote(child.match.group(0))
+                child.sendline(passwd)
+                excute = True
+            else:
+                excute = False
+        if excute or script.lower() == "yes":
+            if stop_after == None and "polling" not in cmd_string and "streaming" not in cmd_string:
+                j_index = child.expect(['.*(%|#|\$)', pexpect.EOF, pexpect.TIMEOUT], timeout=50)
+                if j_index == 1 or j_index == 0:
+                    if "client had error while displaying results" not in child.before:
+                        if j_index == 1:
+                            sleep(5)
+                            result = child.before
+                            testcase_Utils.pNote(result)
+                            #self.ju.pretty_print_json(result)
+                        else:
+                            result = child.after
+                            testcase_Utils.pNote(result)
+                        status = True
+                    else:
+                        status = False
+                        result = child.before
+                        testcase_Utils.pNote(child.before, "error")
+            if stop_after: #For polling or streaming
+                testcase_Utils.pNote("Will Kill the process after {}sec".format(stop_after))
+                child.sendcontrol('C')
+                try:
+                    child.expect([pexpect.EOF, '.*(%|#|\$)'], timeout=int(stop_after))
+                except:
+                    testcase_Utils.pNote("Sending Ctrl+Z")
+                if "client had error while displaying results" not in child.before:
+                    result = child.before.strip().strip("^Z")
+                    testcase_Utils.pNote(result)
+                status = True
+        return status, result, child
+
+    def verify(self, json_object, serach_list):
+        """
+        Verify gNMI output JSON
+        :param json_object:
+        :param serach_list:
+        :return: True or False
+        """
+        status = True
+        if isinstance(json_object, dict):
+            json_str = json.dumps(json_object)
+        else:
+            json_str = json_object

```

Comment:
@safarheen
there is a possibility that this json_object can be None type. Because in case of gNMI_set this is the return from the execuate function in line 92 above. that value of result (which can be None) is passed a json_object to this function.

if we do not intialize json_str to an empty string then the None type will be passed to the regular expression search below and it will fail as expecting string or buffer which was the issue seen by ramya purushotaman as mentioned in the jira ticket.

  File "/home/rpurusho/automation_scripts/gnmi_automation/warriorframework/warrior/Framework/ClassUtils/gnmi_utils_class.py", line 184, in verify

    if re.search(search_pattern, json_str, re.DOTALL):

  File "/local/python/2.7.8/lib/python2.7/re.py", line 142, in search

    return _compile(pattern, flags).search(string)

TypeError: expected string or buffer

Reply:
Addressed